dashboard/lib/pd/foorm/response_processor.rb

```diff
@@ -0,0 +1,163 @@
+        }
+      end
+
+      result = {
+        total_responses: total_responses,
+        breakdown: breakdown
+      }
+
+      # Only include other_answers if the question supports "Other" responses
+      if has_other
+        result[:other_answers] = question_summary['other_answers'] || []
+      end
+
+      result
+    end
+
+    # Process multi-select responses
+    def self.process_multi_select_responses(question_summary, choices, has_other = false)
+      return {} unless question_summary.is_a?(Hash)
+
+      total_respondents = question_summary[:num_respondents] || 0
+      breakdown = {}
+      skip_keys = [:num_respondents, 'other_answers']
+
+      question_summary.each do |choice_key, count|
+        next if skip_keys.include?(choice_key)
+
+        choice_label = choices[choice_key] || choice_key
+        breakdown[choice_key] = {
+          count: count,
+          percentage: calculate_percentage(count, total_respondents, 1),
+          label: choice_label
+        }
+      end
+
+      result = {
+        total_respondents: total_respondents,
+        breakdown: breakdown
+      }
+
+      # Only include other_answers if the question supports "Other" responses
+      if has_other
+        result[:other_answers] = question_summary['other_answers'] || []
+      end
+
+      result
+    end
+
+    # Process rating responses with promoter percentage calculation (for NPS-style 0-10 ratings)
+    def self.process_rating_responses(question_summary, choices)
+      return {} unless question_summary.is_a?(Hash)
+
+      total_responses = question_summary.values.sum
+      breakdown = {}
+      promoter_count = 0
+
+      question_summary.each do |choice_key, count|
+        next if choice_key == 'other_answers'
+
+        choice_label = choices[choice_key] || choice_key
+        breakdown[choice_key] = {
+          count: count,
+          percentage: calculate_percentage(count, total_responses, 1),
+          label: choice_label
+        }
+
+        # Count promoters (ratings >= 7)
+        if choice_key.to_i >= PROMOTER_THRESHOLD
+          promoter_count += count
+        end
+      end
+
+      {
+        total_responses: total_responses,
+        promoter_percentage: calculate_percentage(promoter_count, total_responses),
+        breakdown: breakdown
+      }
+    end
+
+    # Process Likert scale responses with weighted score calculation (for 1-7 agreement scales)
+    def self.process_likert_responses(question_summary, choices)
+      return {} unless question_summary.is_a?(Hash)
+
+      total_responses = question_summary.values.sum
+      breakdown = {}
+      weighted_sum = 0
+      agreement_count = 0 # Count of responses >= 5 (Slightly Agree and above)
+
+      question_summary.each do |choice_key, count|
+        next if choice_key == 'other_answers'
+
+        choice_value = choice_key.to_i
+        choice_label = choices[choice_key] || choice_key
+        weighted_value = LIKERT_WEIGHTS[choice_value] || 0
```

Comment:
If a `weighted_value` somehow has a value not found in `LIKERT_WEIGHTS`, would we want to also ignore it from the `weighted_sum`? Or else missing fields will drag down the average by a lot.